<launch>
    <arg name="class_names_path" default="$(find tam_object_detection)/io/config/wrc_ycb_class_names.txt"/>
    <arg name="weight_path" default="$(find tam_object_detection)/io/models/yolov8-seg_wrc_ycb.engine"/>
    <arg name="device" default="cuda:0"/>
    <arg name="confidence_th" default="0.3"/>
    <arg name="iou_th" default="0.65"/>

    <arg name="camera_info_topic" default="/camera/rgb/camera_info"/>
    <arg name="rgb_topic" default="/camera/rgb/image_raw"/>
    <arg name="use_depth" default="false"/>
    <arg name="depth_topic" default="/camera/depth_registered/image_raw"/>

    <arg name="use_segment" default="true"/>
    <arg name="use_latest_image" default="false"/>
    <arg name="show_tf" default="false"/>
    <arg name="max_distance" default="-1"/>
    <arg name="specific_id" default=""/>

    <node pkg="tam_object_detection" type="yolov8_trt_node.py" name="yolov8_trt_node" output="screen">
        <param name="/class_names_path" type="str" value="$(arg class_names_path)"/>
        <param name="/weight_path" type="str" value="$(arg weight_path)"/>
        <param name="/device" type="str" value="$(arg device)"/>
        <param name="/confidence_th" type="double" value="$(arg confidence_th)"/>
        <param name="/iou_th" type="double" value="$(arg iou_th)"/>

        <param name="/camera_info_topic" type="str" value="$(arg camera_info_topic)"/>
        <param name="/rgb_topic" type="str" value="$(arg rgb_topic)"/>
        <param name="/use_depth" type="bool" value="$(arg use_depth)"/>
        <param name="/depth_topic" type="str" value="$(arg depth_topic)"/>

        <param name="/use_segment" type="bool" value="$(arg use_segment)"/>
        <param name="/use_latest_image" type="bool" value="$(arg use_latest_image)"/>
        <param name="/show_tf" type="bool" value="$(arg show_tf)"/>
        <param name="/max_distance" type="double" value="$(arg max_distance)"/>
        <param name="/specific_id" type="str" value="$(arg specific_id)"/>
    </node>
</launch>
